js/app.js

```diff
@@ -0,0 +1,128 @@
+// Array to store currency rates
+let currencyRates = [];
+
+// Function for adding a new exchange rate
+function handleAddRateForm(event) {
+  event.preventDefault(); // Prevent the default form submission behavior
+
+  // Get the value of the base currency from the input field
+  const baseCurrency = document.getElementById("base-currency").value;
+
+  // Get the value of the target currency from the input field
+  const targetCurrency = document.getElementById("target-currency").value;
+
+  // Get the value of the exchange rate from the input field and convert it to a float
+  const exchangeRate = parseFloat(
+    document.getElementById("exchange-rate").value
+  );
+
+  // Find the rate object for the base currency, or create a new one if it doesn't exist
+  let rateObject = currencyRates.find((rate) => rate.base === baseCurrency);
```

Comment:
You need to clean up the user input for currencies, both `baseCurrency` and `targetCurrency`. See the following screenshot, `eur` and `EUR` should be considered as the same currency:
<img width="381" alt="image" src="https://github.com/l0rians/currency-converter/assets/5467758/b36ad443-808d-42ef-a210-74dd57b80154">

Reply:
Thank you, fixed by <toUpperCase();>

Reply:
Cool. You also need to use `trim()` to avoid having `EUR` and `  EUR   ` considered different currencies.